test/views/declaration/CarrierEoriNumberViewSpec.scala

```diff
@@ -0,0 +1,119 @@
+        val view = createView(form = carrierEoriNumber)
+
+        view
+          .getElementById("Yes")
+          .getElementsByAttribute("checked")
+          .attr("value") mustBe YesNoAnswers.yes
+      }
+
+      "have proper messages for labels" in {
+        messages must haveTranslationFor("declaration.carrierEori.title")
+        messages must haveTranslationFor("declaration.carrierEori.eori.label")
+        messages must haveTranslationFor("declaration.carrierEori.hasEori.empty")
+        messages must haveTranslationFor("declaration.carrierEori.help-item1")
+      }
+
+      "display page title" in {
+        view.getElementsByClass(Styles.gdsPageLegend) must containMessageForElements("declaration.carrierEori.title")
+      }
+
+      "display section header" in {
+        view.getElementById("section-header") must containMessage("declaration.section.2")
+      }
+
+      "display eori question" in {
+        view.getElementsByClass("govuk-label--m") must containMessageForElements("declaration.carrierEori.eori.label")
+        view.getElementById("eori").attr("value") mustBe empty
+      }
+
+      "display 'Back' button that links to 'Exporter Details' page" in {
+
+        val backButton = view.getElementById("back-link")
+
+        backButton must containMessage("site.back")
+        backButton.getElementById("back-link") must haveHref(controllers.declaration.routes.RepresentativeStatusController.displayPage(Mode.Normal))
+      }
+
+      "display 'Save and continue' button" in {
+        val saveButton = view.getElementById("submit")
+        saveButton must containMessage("site.save_and_continue")
+      }
+
+      "display 'Save and return' button on page" in {
+        val saveAndReturnButton = view.getElementById("submit_and_return")
+        saveAndReturnButton must containMessage("site.save_and_come_back_later")
+      }
+
+      "handle invalid input" should {
+        "display errors when all inputs are incorrect" in {
+          val data = CarrierEoriNumber(Some(Eori("123456789")), YesNoAnswers.yes)
+          val form = CarrierEoriNumber.form().fillAndValidate(data)
+          val view = createView(form = form)
+
+          view must haveGovukGlobalErrorSummary
+          view must containErrorElementWithTagAndHref("a", "#eori")
+          view must containErrorElementWithMessageKey("declaration.eori.error.format")
+        }
+
+        "display errors when eori contains special characters" in {
+          val data = CarrierEoriNumber(eori = Some(Eori("12#$%^78")), hasEori = YesNoAnswers.yes)
+          val form = CarrierEoriNumber.form().fillAndValidate(data)
+          val view = createView(form = form)
+
+          view must haveGovukGlobalErrorSummary
+          view must containErrorElementWithTagAndHref("a", "#eori")
+          view must containErrorElementWithMessageKey("declaration.eori.error.format")
+        }
+      }
```

Comment:
Nit: This is something I am always unsure about - whether view specs should test for different possible errors, or maybe it is enough to just check an error. If there is error summary, error text near the field etc., and leave the responsibility to test for all errors to form specs.
This is not a problem, but it's just I've always had this inner conflict.

Reply:
Well not knowing the code that well I just copied what the ConsignorEoriNumberViewSpec did. If this is something we should change then probably should do it for all view specs.

Reply:
Agree with you. But sadly, taking this approach we are going to probably never actually do it ;-P